### ISSUE TITLE
Wrap IndexSignatureDeclaration

### ISSUE THREAD

**Issue by kjpou1** [dsherret/ts-simple-ast#242]
Would open up navigation without resorting to ts.IndexSignature <=> ts-simple-ast casting.

example with casting:

```
function foo (node : sast.Node)
{
   const indexSignatureDeclaration = <ts.IndexSignatureDeclaration> node.compilerNode;
   const isitreally = ts.isIndexSignatureDeclaration(indexSignatureDeclaration);
   const isitreallyreally = ts.isIndexSignatureDeclaration(node.compilerNode);
   console.log((<any>ts).SyntaxKind[indexSignatureDeclaration.kind]);

   for (let i = 0, n = indexSignatureDeclaration.parameters.length; i < n; i++) {
     const param = indexSignatureDeclaration.parameters[i];
     console.log((<any>ts).SyntaxKind[param.kind]);
   }

   console.log((<any>ts).SyntaxKind[indexSignatureDeclaration.type.kind]);
}
```

**Comment by dsherret** [comment]
I was surprised I never implemented this. Implemented now in 6.11.0.

**Comment by kjpou1** [comment]
Works like a charm.  Thanks.
